pageserver/ctl/src/layers.rs

```diff
@@ -70,11 +79,47 @@ fn read_delta_file(path: impl AsRef<Path>) -> Result<()> {
         },
     )?;
     let mut cursor = BlockCursor::new(&file);
+
+    let mut result = vec![];
     for (k, v) in all {
         let value = cursor.read_blob(v.pos())?;
-        println!("key:{} value_len:{}", k, value.len());
+        result.push((k, value.len()));
     }
     // TODO(chi): special handling for last key?
+    Ok(result)
+}
+
+// We divide the entire i128 keyspace into pre-assigned fixed segments,
+// 8MB each. Group keys by segment, and report segment size for each.
+//
+// 8MB is chosen as the segment size because we're unlikely to make
+// s3 partial downloads smaller than 8KB (due to cost). So summarizing
+// layer metadata in 8KB segments could be enough to generate good test
+// data for write amplification tests.
+fn read_delta_segments(path: impl AsRef<Path>) -> Result<Vec<(i128, usize)>> {
+    fn key_to_segment(key: &Key) -> i128 {
+        key.to_i128() >> 26
```

Comment:
Where is the `26` from? I thought it is something like 8MB, but should it be `>>23` instead?

Reply:
better to add comments here 🤪 also for delta layer, I don't think the value is always of the size 8KB? This only holds for image layers?

Reply:
> A page is 8KB

A better way to say it is "key X points at 8KB*X". Size of delta entries doesn't matter, it's a fixed partitioning based on the address

Reply:
Let me know if some rewrite of the docstring would make it better